app/client/cli/servicer.go

```diff
@@ -0,0 +1,284 @@
+		{
+			Use:   "Relay <servicer> <application> <relayChainID> <payload>",
+			Short: "Relay <servicer> <application> <relayChainID> <payload>",
+			Long: `Sends a trustless relay using <payload> as contents, to the specified active <servicer> in the the <application>'s session.
+Will prompt the user for the *application* account passphrase`,
+			Aliases: []string{},
+			Args:    cobra.ExactArgs(4),
+			RunE: func(cmd *cobra.Command, args []string) error {
+				servicerAddr := args[0]
+				applicationAddr := args[1]
+				chain := args[2]
+				relayPayload := args[3]
+
+				// TODO: (SUGGESTION) refactor to decouple the client logic from the CLI/command
+				pk, err := getPrivateKey(applicationAddr)
+				if err != nil {
+					return fmt.Errorf("error getting application's private key: %w", err)
+				}
+
+				session, servicer, err := fetchServicer(cmd.Context(), applicationAddr, chain, servicerAddr)
+				if err != nil {
+					return fmt.Errorf("error getting servicer for the relay: %w", err)
+				}
+
+				relay, err := buildRelay(relayPayload, pk, session, servicer)
+				if err != nil {
+					return fmt.Errorf("error building relay from payload: %w", err)
+				}
+
+				fmt.Printf("sending trustless relay for %s to %v with payload: %s\n", applicationAddr, servicer, relayPayload)
+
+				resp, err := sendTrustlessRelay(cmd.Context(), servicer.ServiceUrl, relay)
+				if err != nil {
+					return err
+				}
+
+				fmt.Printf("HTTP status code: %d\n", resp.HTTPResponse.StatusCode)
+				fmt.Println("Response: ", resp.JSON200)
+
+				return nil
+			},
+		},
+	}
+
+	return cmds
+}
+
+// TODO: (QUESTION): do we need/want a cli subcommand for fetching servicers?
+
+// fetchServicer returns the servicer specified by the <servicer> argument.
+// It validates the following conditions:
+//
+//	A. The <application> argument is the address of an active application
+//	B. The <servicer> is the address of a servicer that is active in the application's current session.
+//
+// TODO: (SUGGESTION) use a package-internal interface for servicer and application?
+// TODO: (SUGGESTION) use a struct as input to combine all fields (same for output)
+func fetchServicer(ctx context.Context, appAddress, chain, servicerAddress string) (rpc.Session, rpc.ProtocolActor, error) {
+	// TECHDEBT: cache session data
+	session, err := getCurrentSession(ctx, appAddress, chain)
+	if err != nil {
+		return rpc.Session{}, rpc.ProtocolActor{}, fmt.Errorf("Error getting servicer: %w", err)
+	}
+
+	var (
+		servicer rpc.ProtocolActor
+		found    bool
+	)
+	// TODO: a map may be a better choice for storing servicers
+	for _, s := range session.Servicers {
+		if s.Address == servicerAddress {
+			servicer = s
+			found = true
+			break
+		}
+	}
+
+	// TODO: cover with unit tests
+	if !found {
+		return rpc.Session{}, rpc.ProtocolActor{}, fmt.Errorf("Error getting servicer: address %s does not match any servicers in the session", servicerAddress)
+	}
+
+	// TODO: cover with unit tests
```

Comment:
You shouldn't need to do this. getCurrentSession accepts a chain, so the protocol wouldn't return a servicer unless it supports the chain for that height.

Reply:
Thanks. Fixed.